### PR TITLE
Updated translator to access new caching features of translation component

### PR DESCRIPTION
**Pull request by CarsonF**
I updated Translator and TranslationServiceProvider to accept a cache directory and debug boolean.

### CODE REVIEW

File: src/Silex/Provider/TranslationServiceProvider.php [hunk 1]
```diff
@@ -53,6 +58,8 @@ public function register(Application $app)
             return new MessageSelector();
         });
 
+        $app['translator.cache_dir'] = null;
+        $app['translator.debug'] = $app['debug'];
```
Comment (stof) [inline src/Silex/Provider/TranslationServiceProvider.php]: this should use a lazy definition to allow it to take into account changes to the global debug mode done after the registration of the provider

File: src/Silex/Provider/TranslationServiceProvider.php [hunk 2]
```diff
@@ -53,6 +58,10 @@ public function register(Application $app)
             return new MessageSelector();
         });
 
+        $app['translator.cache_dir'] = null;
+        $app['translator.debug'] = function() use ($app) {
```
Comment (stof) [inline src/Silex/Provider/TranslationServiceProvider.php]: this should use ``function ($app)`` to be consistent with the way the translator service is defined

Comment (CarsonF) [inline src/Silex/Provider/TranslationServiceProvider.php]: :O I didn't realize that worked

**Comment by fabpot** [comment]
#1124 has been merged now.

**Pull request closed without merging**